servercfg/config.go

```diff
@@ -423,7 +476,47 @@ func (cfg *DoltgresConfig) ToSqlServerConfig() servercfg.ServerConfig {
 	return cfg
 }
 
-func ReadConfigFromYamlFile(fs filesys.Filesys, configFilePath string, overrides map[string]string) (*DoltgresConfig, error) {
+// DefaultServerConfig creates a `*DoltgresConfig` that has all of the options set to their default values. Used when
+// no config.yaml file is provided.
+func DefaultServerConfig() *DoltgresConfig {
+	dataDir := "."
+	homeDir, err := env.GetCurrentUserHomeDir()
+	if err == nil {
+		dataDir = filepath.Join(homeDir, DOLTGRES_DATA_DIR_DEFAULT)
+	}
+	
+	return &DoltgresConfig{
+		LogLevelStr:       ptr(string(DefaultLogLevel)),
+		EncodeLoggedQuery: ptr(DefaultEncodeLoggedQuery),
+		BehaviorConfig: &DoltgresBehaviorConfig{
+			ReadOnly:              ptr(DefaultReadOnly),
+			DoltTransactionCommit: ptr(DefaultDoltTransactionCommit),
+		},
+		UserConfig: &DoltgresUserConfig{
+			Name:     ptr(DefaultUser),
+			Password: ptr(DefaultPass),
+		},
+		ListenerConfig: &DoltgresListenerConfig{
+			HostStr:                 ptr(DefaultHost),
+			PortNumber:              ptr(DefaultPort),
+			ReadTimeoutMillis:       ptr(uint64(DefaultTimeout)),
+			WriteTimeoutMillis:      ptr(uint64(DefaultTimeout)),
+			AllowCleartextPasswords: ptr(DefaultAllowCleartextPasswords),
+		},
+		PerformanceConfig: &DoltgresPerformanceConfig{QueryParallelism: ptr(DefaultQueryParallelism)},
+		
+		DataDirStr:        ptr(dataDir),
+		CfgDirStr:         ptr(filepath.Join(DefaultDataDir, DefaultCfgDir)),
+		PrivilegeFile:     ptr(filepath.Join(DefaultDataDir, DefaultCfgDir, DefaultPrivilegeFilePath)),
+		BranchControlFile: ptr(filepath.Join(DefaultDataDir, DefaultCfgDir, DefaultBranchControlFilePath)),
+	}
+}
+
+func ptr[T any](t T) *T {
```

Comment:
I understand why this is here (can't take a pointer to an expression's result without first assigning it to a variable), but we should add a comment for why it exists, cause it seems really weird without knowing that.

Also comments on all the functions would make me happy, but it's not necessary lol

Reply:
Added a comment for prt